src/App.tsx

```diff
@@ -16,34 +16,95 @@ export const goodsFromServer = [
 ];
 
 export const App: React.FC = () => {
+  const [activeSort, setActiveSort] = useState<string>('all');
+  const [isReversed, setIsReversed] = useState<boolean>(false);
+
+  enum SortStatus {
+    All = 'all',
+    Alphabetical = 'alphabetical',
+    Length = 'length',
+  }
+
+  function getSort(): string[] {
+    const sorted = [...goodsFromServer];
+
+    if (activeSort === SortStatus.Alphabetical) {
+      sorted.sort((a, b) => a.localeCompare(b));
+    } else if (activeSort === SortStatus.Length) {
+      sorted.sort((a, b) => a.length - b.length);
+    }
+
+    if (isReversed) {
+      sorted.reverse();
+    }
+
+    return sorted;
+  }
+
+  const resetOnClick = () => {
```

Comment:
mind the naming (usually handlers start from "handle")

```suggestion
  const handleResetClick = () => {
```